packages/medusa/src/scripts/migrate-to-pricing-module.ts

```diff
@@ -0,0 +1,232 @@
+          const updateData: PricingTypes.UpdatePriceListDTO = {
+            id: priceList.id,
+            title: priceList.name,
+          }
+
+          if (priceList?.customer_groups?.length) {
+            updateData.rules = {
+              customer_group_id: priceList.customer_groups.map(({ id }) => id),
+            }
+          }
+
+          return updateData
+        })
+      )
+
+      await pricingModuleService.addPriceListPrices(
+        priceListsToUpdate.map((priceList) => {
+          return {
+            priceListId: priceList.id,
+            prices: priceList.prices
+              .filter((price) =>
+                variantIdPriceSetIdMap.has(price.variants?.[0]?.id)
+              )
+              .map((price) => {
+                return {
+                  price_set_id: variantIdPriceSetIdMap.get(
+                    price.variants?.[0]?.id
+                  )!,
+                  currency_code: price.currency_code,
+                  amount: price.amount,
+                  min_quantity: price.min_quantity,
+                  max_quantity: price.max_quantity,
+                }
+              }),
+          }
+        })
+      )
+    }
+
+    if (priceListsToCreate.length) {
+      await pricingModuleService.createPriceLists(
+        priceListsToCreate.map(
+          ({ name: title, prices, customer_groups, ...priceList }) => {
+            const createData: PricingTypes.CreatePriceListDTO = {
+              ...priceList,
+              starts_at: priceList.starts_at?.toISOString(),
+              ends_at: priceList.ends_at?.toISOString(),
+              title,
+            }
+
+            if (customer_groups?.length) {
+              createData.rules = {
+                customer_group_id: customer_groups.map(({ id }) => id),
+              }
+            }
+
+            if (prices?.length) {
+              createData.prices = prices.map((price) => {
+                return {
+                  price_set_id: variantIdPriceSetIdMap.get(
+                    price.variants?.[0]?.id
+                  )!,
+                  currency_code: price.currency_code,
+                  amount: price.amount,
+                  min_quantity: price.min_quantity,
+                  max_quantity: price.max_quantity,
+                }
+              })
+            }
+
+            return createData
+          }
+        )
+      )
+    }
+
+    offset += corePriceLists.length
+
+    console.log(`Processed ${offset} of ${totalCount}`)
+  }
+}
+
+const ensureCurrencies = async (container: AwilixContainer) => {
+  const currenciesService: CurrencyService =
+    container.resolve("currencyService")
+
+  const pricingModuleService: IPricingModuleService = container.resolve(
+    "pricingModuleService"
+  )
+
+  const [coreCurrencies, totalCurrencies] =
+    await currenciesService.listAndCount({}, {})
+
+  const moduleCurrencies = await pricingModuleService.listCurrencies(
+    {},
+    { take: 100000 }
```

Comment:
question: maybe we should allow null to be passed in the take, skip and not use the default if null is provided? in the build-query

something like `(Number.isInteger(config.take) || config.take === null) ? config.take : 15` same for the skip